VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+
+        * Mature modules for which we guarantee a stable public API will be versioned
+          with a major version greater than `v0`.
+            * The decision to make a module stable will be made on a case-by-case
+              basis by the maintainers of this project.
+        * Experimental modules will start their versioning at `v0.0.0` and will
+          increment their minor version when backwards incompatible changes are
+          released and increment their patch version when backwards compatible
+          changes are released.
+        * All stable modules that use the same major version number will use the
+          same entire version number.
+            * Stable modules may be released with an incremented minor or patch
+              version even though that module has not been changed, but rather so
+              that it will remain at the same version as other stable modules that
+              did undergo change.
+            * When an experimental module becomes stable a new stable module version
+              will be released and will include this now stable module. The new
+              stable module version will be an increment of the minor version number
+              and will be applied to all existing stable modules as well as the newly
+              stable module being released.
+* Versioning of the associated [contrib
+  repository](https://github.com/open-telemetry/opentelemetry-collector-contrib) of
+  this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
+        * If a module is version `v2` or higher, the
+          major version of the module must be included as a `/vN` at the end of the
+          module paths used in `go.mod` files (e.g., `module
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2`, `require
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2 v2.0.1`) and in the
+          package import path (e.g., `import
+          "github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2"`). This includes
+          the paths used in `go get` commands (e.g., `go get
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2@v2.0.1`.  Note there
+          is both a `/v2` and a `@v2.0.1` in that example. One way to think about
+          it is that the module name now includes the `/v2`, so include `/v2`
+          whenever you are using the module name).
+        * If a module is version `v0` or `v1`, do not include the major version
+          in either the module path or the import path.
+    * In addition to public APIs, telemetry produced by stable instrumentation
```

Comment:
What `instrumentation` does this refer to? The telemetry produced by receivers such as hostmetrics? Or the self-telemetry of components (e.g. what is produced via obsreport today)?

Reply:
I will remove this. It is vestigial from the Go client library policy and, even there, I'm not sure we're ready to make any such guarantees.  Schema versioning for the semantic conventions will help, but I think it's still an open question as to what it means for telemetry to be "stable".